src/Rules/Zend/ServiceManagerGetMethodCallRule.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ZendPhpStan\Rules\Zend;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ObjectType;
+use Zend\Mvc\Controller\ControllerManager;
+use Zend\ServiceManager\PluginManagerInterface;
+use Zend\View\HelperPluginManager;
+use ZendPhpStan\Type\Zend\ObjectServiceManagerType;
+use ZendPhpStan\Type\Zend\ServiceManagerLoader;
+
+final class ServiceManagerGetMethodCallRule implements Rule
+{
+    /**
+     * @var ServiceManagerLoader
+     */
+    private $serviceManagerLoader;
+
+    /**
+     * @var array
+     */
+    private $knownUnmappedAliasToClassServices = [
+        ControllerManager::class   => 'ControllerManager',
+        HelperPluginManager::class => 'ViewHelperManager',
```

Comment:
@weierophinney I'm writing this extension to https://github.com/phpstan/phpstan that checks the correct usage of configured services and plugins.

Everything works flawlessly, expect for services not mapped with their class to the common name, like this two.
So for example if a user writes this:
```php
/** @var \Zend\View\HelperPluginManager $viewHelperManager */
$viewHelperManager = $whateverGetter();
```
Without these two lines I'm commenting at, I wouldn't be able to recognize the service.

Looking at the source code of each Zend component, it appears that there isn't any specific rule you've redacted to enforce `xxxPlugin::class => 'XxxPluginManager'` map.

Another example, `ControllerPluginManager` is mapped:

https://github.com/zendframework/zend-mvc/blob/9f092ff24945f208047d99dff81f67de8862b5f4/src/Service/ServiceListenerFactory.php#L54

`ControllerManager` isn't mapped:

https://github.com/zendframework/zend-mvc/blob/master/src/Service/ServiceListenerFactory.php#L66

How should I handle this topic?
Do I have to assume that there will be no rule, and thus should I maintain this custom list, or maybe it is a bug to be reported in the related component?